src/app/clusters/door-lock-server/door-lock-server.cpp

```diff
@@ -140,30 +143,115 @@ bool emberAfDoorLockClusterLockDoorCallback(chip::app::CommandHandler * commandO
                                             const chip::app::ConcreteCommandPath & commandPath,
                                             const chip::app::Clusters::DoorLock::Commands::LockDoor::DecodableType & commandData)
 {
-    emberAfDoorLockClusterPrintln("Received Lock Door command (not implemented)");
+    emberAfDoorLockClusterPrintln("Received Lock Door command");
+    bool success = false;
 
-    // TODO: Implement door locking by calling emberAfPluginDoorLockOnDoorLockCommand
+    chip::EndpointId endpoint = commandPath.mEndpointId;
 
-    emberAfSendImmediateDefaultResponse(EMBER_ZCL_STATUS_SUCCESS);
-    return true;
+    bool require_pin = false;
+    Attributes::RequirePINforRemoteOperation::Get(endpoint, &require_pin);
+
+    if(require_pin)
+    {
+        if(!commandData.pinCode.HasValue())
+        {
+            success = false; // Just to be explicit. success == false at this point anywyay
+        }
+        else
+        {
+            const char *cmd_pin = reinterpret_cast<const char*>(commandData.pinCode.Value().data());
```

Comment:
No, you can't do that.  The value is not null-terminated, and is not a char string anyway.  The might be no null anywhere around, or it might contain multiple nulls.

Reply:
Fixed